web/src/integTest/java/com/microsoft/applicationinsights/azuretest/CompareTelemetryWithAzureExportTests.java

```diff
@@ -0,0 +1,342 @@
+        String address       = json.getJSONArray("request").getJSONObject(0).getString("url");
+        Integer port         = json.getJSONArray("request").getJSONObject(0).getJSONObject("urlData").getInt("port");
+        Integer responseCode = json.getJSONArray("request").getJSONObject(0).getInt("responseCode");
+        String userAgent     = json.getJSONObject("context").getJSONObject("device").getString("userAgent");
+
+        JSONArray parameters = json.getJSONArray("request").getJSONObject(0).getJSONObject("urlData").getJSONArray("queryParameters");
+        Hashtable<String, String> queryParameters = new Hashtable<String, String>();
+        for (int i = 0; i < parameters.length(); ++i) {
+            String name  = parameters.getJSONObject(i).getString("parameter");
+            String value = parameters.getJSONObject(i).getString("value");
+            queryParameters.put(name, value);
+        }
+
+        TelemetryItem telemetryResult = new TelemetryItem(DocumentType.Requests, queryParameters.get("requestId"));
+        telemetryResult.setProperty("uri", address);
+        telemetryResult.setProperty("port", port.toString());
+        telemetryResult.setProperty("responseCode", responseCode.toString());
+        telemetryResult.setProperty("userAgent", userAgent);
+
+        for (String key : queryParameters.keySet()) {
+            telemetryResult.setProperty("queryParameter." + key, queryParameters.get(key));
+        }
+
+        return telemetryResult;
+    }
+
+    /**
+     * Converts JSON object to PerformanceCounter TelemetryItem
+     * @param json The JSON object
+     * @return A TelemetryItem
+     */
+    private TelemetryItem getPerformanceCounterTelemetryItemFromJson(JSONObject json) throws JSONException {
+        System.out.println("Converting JSON object to telemetry item with document type " + DocumentType.PerformanceCounters.toString());
+        String category = json.getJSONArray("performanceCounter").getJSONObject(0).getString("categoryName");
+        String instance = json.getJSONArray("performanceCounter").getJSONObject(0).getString("instanceName");
+
+        TelemetryItem telemetryResult = new TelemetryItem(DocumentType.PerformanceCounters, ""); // TODO when implementing the usage of performance counters test, think of logic for id.
+        telemetryResult.setProperty("category", category);
+        telemetryResult.setProperty("instance", instance);
+
+        return telemetryResult;
+    }
+
+    /**
+     * Tests if the expected telemetry exists in the real telemetry from AppInsights
+     * @param realTelemetryItems The real telemetry
+     * @param expectedTelemetryItems The expected telemetry
+     * @return A collection of expected telemetry items that were not found in the real telemetry. If all items exists, returns empty collection.
+     */
+    private ArrayList<TelemetryItem> getMissingExpectedTelemetry(ArrayList<TelemetryItem> realTelemetryItems,
```

Comment:
We should be able to check bi-directional inclusion. Therefore please change this method name and local variables to be symmetric and not assume that one of the collections is the real and the other is the expected.